src/Symfony/Component/AssetMapper/AssetMapperDevServerSubscriber.php

```diff
@@ -160,4 +156,32 @@ private function getMediaType(string $path): ?string
 
         return $this->extensionsMap[$extension] ?? null;
     }
+
+    private function findAssetFromCache(string $pathInfo): ?MappedAsset
+    {
+        $cachedAsset = $this->cacheMapCache->getItem(hash('xxh128', $pathInfo));
+        $asset = $cachedAsset->isHit() ? $this->assetMapper->getAsset($cachedAsset->get()) : null;
+
+        if (null !== $asset && $asset->getPublicPath() === $pathInfo) {
+            return $asset;
+        }
+
+        // we did not find a match
+        $asset = null;
+        foreach ($this->assetMapper->allAssets() as $assetCandidate) {
```

Comment:
if not, can't we make assAssets return an iterator instead of an array, so that we don't have to load everything in memory?

Reply:
> would there be a way to list only the public paths, and skip loading the full assets in memory?

MAYBE :p. To generate the public paths, something needs to needs to load all the `MappedAsset[]` (as public path depends on contents for the digest). `AssetMapper` *could* have some sort of `->allPublicPaths()` or even `findByPublicPath()` where we create that "public path map" and cache it (I "think" we would not need to worry about invalidation thanks to the digest: i.e. public path changes when file changes). It's basically what I'm doing here, which I realize is kind of an odd spot, but it's also only needed here.

> if not, can't we make allAssets return an iterator instead of an array, so that we don't have to load everything in memory?

Definitely! Though... if we have 200 assets and the one we want is number 190, it's still not ideal. It would be simpler, but each request would, on average, I guess still load 50% of the assets?